.github/workflows/build.yml

```diff
@@ -15,90 +15,160 @@
 #
 name: Elyra Tests
 on:
-  push:
-    branches: '*'
-  pull_request:
-    branches: '*'
+  push: # all branches
+  pull_request: # all branches
   schedule:
     # once a day at 3 am (UTC) (7 pm (PST))
     - cron:  '0 3 * * *'
 
+env:
+  FORCE_COLOR: true
+
 jobs:
-  build:
-    runs-on: ${{ matrix.os }}
+  prepare-yarn-cache:
+    name: Prepare Cache
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "*"
+      - uses: actions/cache@v2
+        with:
+          path: |
+            node_modules
+            */*/node_modules
+            /home/runner/.cache/Cypress
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+      - name: Install
+        run: yarn install --frozen-lockfile
+
+  lint-server:
+    name: Lint Server
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install
+        run: |
+          python -m pip install --upgrade pip
```

Comment:
We already have these steps using make, could we avoid duplication here and use the make target.

Reply:
The `install-server` target will end up running flake as a side effect. Can we rewire the `install-xxx` targets to not do linting? Maybe only lint when running the full `make install`? Or create new make targets that only do installing and `install-xxx` calls those and lint?

Reply:
I'd be fine with that.  I would like to make sure we do linting on test-related targets, however.  The key being that linting gets performed prior to PR creation and is something that tends to be forgotten.

Reply:
Do you think we should just remove linting from install all together and just have it run on test?

Reply:
I guess the confusing thing here is that `make install` is actually _the_ build step and linting does make sense prior to building.  Perhaps we could introduce explicit build-xxx targets that include linting, although I'm not sure that helps from a CI perspective.

I guess I should ask, what is the issue with `flake8` running prior to building the server?

Reply:
I'm fine with linting before building the server locally. In CI linting is separated into its own job, so it would end up linting in the linting job twice and then also in every job where the server is built. If linting fails then none of the other tests would run

Reply:
The more I'm thinking about this, the more I realize that our make targets try to group everything into one giant command, where the CI is trying to separate everything into much smaller distinct steps

Reply:
> The more I'm thinking about this, the more I realize that our make targets try to group everything into one giant command
And why is that a problem? In terms of lint, the only additional thing is "preparing/installing" dependencies that will be used in a future step... In the worst case, we separate in make, but we also don't want contributors to have to issue single commands for every single task.